src/jit/lsra.cpp

```diff
@@ -5667,16 +5666,30 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
+        RegRecord* physRegToUnassign = availablePhysRegInterval;
+#ifdef _TARGET_ARM_
+        if (!isUnassigned(availablePhysRegInterval, currentInterval))
```

Comment:
Also, with this change you are calling this method twice, rather than cashing the information that it is unassigned. It's probably not a big deal, but this is a hot method.